traefik.go

```diff
@@ -209,29 +209,36 @@ func main() {
 	log.Info("Shutting down")
 }
 
-func createTLSConfig(certFile string, keyFile string) (*tls.Config, error) {
+// creates a TLS config that allows terminating HTTPS for multiple domains using SNI
+func createTLSConfig(certs []Certificate) (*tls.Config, error) {
 	config := &tls.Config{}
 	if config.NextProtos == nil {
 		config.NextProtos = []string{"http/1.1"}
 	}
 
 	var err error
-	config.Certificates = make([]tls.Certificate, 1)
-	if len(certFile) > 0 && len(keyFile) > 0 {
-		config.Certificates[0], err = tls.LoadX509KeyPair(certFile, keyFile)
-		if err != nil {
-			return nil, err
+	if len(certs) > 0 {
```

Comment:
You have a point there. I tried to to stick as closely as possible with the existing code structure because some maintainers get grumpy when they see refactoring in a feature patch.  :wink: